src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/Target_ARM/ARMEmitter.cs

```diff
@@ -32,12 +61,44 @@ public void EmitPUSH(Register reg)
             Builder.EmitShort((short)(0x0d04 + ((byte)reg << 12)));
         }
 
+        // pop reg
+        public void EmitPOP(Register reg)
+        {
+            Builder.EmitByte(0x5d);
+            Builder.EmitByte(0xf8);
+            Builder.EmitShort((short)(0x0b04 + ((byte)reg << 12)));
+        }
+
         // mov reg, reg
+        // reg range: 0-7
```

Comment:
I checked, it's strange, in official documentation from arm infocenter only 3 bits for Rd and 4 for Rm, but it works with higher registers. ldr for example exactly uses r0-r7 and has 3 bits. Ok, thanks